### ISSUE TITLE
Electron 11.2 Segfaults

### ISSUE THREAD

**Issue by drewmnoel** [jiahaog/nativefier#1101]
**Homework**

- [X] I looked at `nativefier --help` and https://github.com/jiahaog/nativefier/blob/master/docs/api.md
- [X] I searched existing issues, open & closed. Yes, my bug is new.
- [X] I'm using the latest version available at https://github.com/jiahaog/nativefier/releases

**Bug description**

The current 11-x-y of Electron [has a known issue](https://github.com/electron/electron/issues/27303) where the application will segfault if the icon is set to "undefined". Until a fix is merged in upstream, nativefier applications from icon-less sites will crash on macOS and Linux with the default of 11.2. Electron 11.1.1 works fine still, though.

**Steps to reproduce**

```
nativefier --verbose https://example.com
Running in verbose mode! This will produce a mountain of logs and is recommended only for troubleshooting or if you like Shakespeare.

Performing async options post-processing.
Inferring user agent for electron 11.2.0 / linux
Grabbing electron<->chrome versions file from https://atom.io/download/atom-shell/index.json
Inferring icon for https://example.com/ on linux
Inferring icon from store for https://example.com/ on linux
Allowed icon formats when building for linux: [ '.png', '.ico' ]
Inferring name for https://example.com/
Got 249 icons from gitcloud
Max icon match score: 0
No relevant icon in store.
Trying to extract a .png icon from the page.
Fetched 1.2 kb page at https://example.com/
Inferred title: Example Domain
Sanitized filename for Example Domain : ExampleDomain
Associated electron v11.2.0 to chrome v87.0.4280.141
Given chrome 87.0.4280.141 on linux, using user agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.141 Safari/537.36

Preparing Electron app...
Copying electron app from /usr/lib/node_modules/nativefier/app to /tmp/nativefier-23-42-48-app--84117-WC7yoJEiKUP5
Writing app config to /tmp/nativefier-23-42-48-app--84117-WC7yoJEiKUP5/nativefier.json
No files to inject, skipping copy.
Updating /tmp/nativefier-23-42-48-app--84117-WC7yoJEiKUP5/package.json 'name' field to exampledomain-nativefier-182cce

Converting icons...
Option "icon" not set, skipping icon conversion.
Copying icons if necessary
No icon specified in options; aborting

Packaging... This will take a few seconds, maybe minutes if the requested Electron isn't cached yet...
  electron-packager Electron Packager 15.2.0
  electron-packager Node v15.6.0
  electron-packager Host Operating system: linux 5.8.14-arch1-1 (x64) +0ms
  electron-packager Packager Options: {"arch":"x64","asar":false,"darwinDarkModeSupport":false,"dir":"/tmp/nativefier-23-42-48-app--84117-WC7yoJEiKUP5","electronVersion":"11.2.0","name":"ExampleDomain","out":"/tmp","overwrite":true,"platform":"linux","targetUrl":"https://example.com/","tmpdir":false,"win32metadata":{}} +0ms
  electron-packager Target Platforms: linux +0ms
  electron-packager Target Architectures: x64 +0ms
  electron-packager Inferring appVersion from version in /tmp/nativefier-23-42-48-app--84117-WC7yoJEiKUP5/package.json +0ms
  electron-packager Application name: ExampleDomain +7ms
  electron-packager Target Electron version: 11.2.0 +0ms
  electron-packager Ignored path regular expressions: [
  '/package-lock\\.json$',
  '/yarn\\.lock$',
  '/\\.git($|/)',
  '/node_modules/\\.bin($|/)',
  '\\.o(bj)?$',
  '/tmp/electron-packager'
] +0ms
  electron-packager Downloading Electron with options {"platform":"linux","arch":"x64","version":"11.2.0","artifactName":"electron"} +0ms
Packaging app for platform linux x64 using electron v11.2.0
  electron-packager Creating /tmp/linux-x64-template +3ms
  electron-packager Extracting /home/dnoel/.cache/electron/httpsgithub.comelectronelectronreleasesdownloadv11.2.0electron-v11.2.0-linux-x64.zip/electron-v11.2.0-linux-x64.zip to /tmp/linux-x64-template +0ms
  electron-packager Initializing app in /tmp/ExampleDomain-linux-x64 from /tmp/linux-x64-template template +0ms
  electron-packager Ignored paths based on the out param: [
  '/tmp/ExampleDomain-darwin-x64',
  '/tmp/ExampleDomain-darwin-arm64',
  '/tmp/ExampleDomain-linux-ia32',
  '/tmp/ExampleDomain-linux-x64',
  '/tmp/ExampleDomain-linux-armv7l',
  '/tmp/ExampleDomain-linux-arm64',
  '/tmp/ExampleDomain-linux-mips64el',
  '/tmp/ExampleDomain-mas-x64',
  '/tmp/ExampleDomain-mas-arm64',
  '/tmp/ExampleDomain-win32-ia32',
  '/tmp/ExampleDomain-win32-x64',
  '/tmp/ExampleDomain-win32-arm64'
] +2s
  electron-packager Renaming electron to ExampleDomain in /tmp/ExampleDomain-linux-x64 +54ms

Finalizing build...
App built to /tmp/ExampleDomain-linux-x64 , move it wherever it makes sense for you and run the contained executable file (prefixing with ./ if necessary)
Menu/desktop shortcuts are up to you, because Nativefier cannot know where you're going to move the app. Search for "linux .desktop file" for help, or see https://wiki.archlinux.org/index.php/Desktop_entries
```

**Expected behavior**

Running `/tmp/ExampleDomain-linux-x64/ExampleDomain` creates a new Electron page showing the Example domain.

**Actual behavior**

`[1]    93789 segmentation fault (core dumped)  /tmp/ExampleDomain-linux-x64/ExampleDomain`

**Context**

 - Nativefier: 42.1.0
 - Node.js: v15.6.0
 - Npm: 6.14.11
 - OS: Arch Linux

**Comment by qoomon** [comment]
got the same problem

OS: macOS Big Sur
Node.js: v15.6.0
Electron: v11.2.0
Nativefier: 42.1.0

**Comment by ronjouch** [comment]
@drewmnoel thanks for the heads up! Fixed with a dumb revert to default to 11.1.1 (see commit message 10f7fed2902bba92f252ca1f30fc4177ca854837 for why I'm not doing anything more complex:

> Not doing anything more complicated (adding macOS-specific code
> or adding code always passing an icon), let's instead wait for
> Electron to fix the issue.

Cutting a release too, it will be 42.2.0

**Comment by ronjouch** [comment]
This is now fixed in Electron: https://github.com/electron/electron/issues/27303#issuecomment-767101481 , https://github.com/electron/electron/pull/27478 . I'll bump Nativefier to the upcoming 11.x.y release when it's released.
